src/Renci.SshNet/ISftpClient.cs

```diff
@@ -720,7 +719,29 @@ public interface ISftpClient : IDisposable
         /// <exception cref="SftpPermissionDeniedException">Permission to list the contents of the directory was denied by the remote host. <para>-or-</para> A SSH command was denied by the server.</exception>
         /// <exception cref="SshException">A SSH error where <see cref="Exception.Message" /> is the message from the remote host.</exception>
         /// <exception cref="ObjectDisposedException">The method was called after the client was disposed.</exception>
+#if NETSTANDARD2_1_OR_GREATER
+        [Obsolete("Use EnumerateDirectoryAsync()")]
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+#endif
         Task<IEnumerable<SftpFile>> ListDirectoryAsync(string path, CancellationToken cancellationToken);
```

Comment:
I'd prefer to only expose the **IAsyncEnumerable** version (with **ListDirectoryAsync** as name), even if that means it's .NET Standard 2.1 (or higher) only.
Is the **NETSTANDARD2_1_OR_GREATER** symbol also defined for .NET 5.0 or 6.0?